src/net/tcp.rs

```diff
@@ -89,18 +93,26 @@ impl TcpStream {
     /// `net2::TcpBuilder` to configure a socket and then pass its socket to
     /// `TcpStream::connect_stream` to transfer ownership into mio and schedule
     /// the connect operation.
+    #[cfg(not(target_os = "redox"))]
     pub fn connect(addr: &SocketAddr) -> io::Result<TcpStream> {
         let sock = match *addr {
             SocketAddr::V4(..) => TcpBuilder::new_v4(),
             SocketAddr::V6(..) => TcpBuilder::new_v6(),
         }?;
         // Required on Windows for a future `connect_overlapped` operation to be
         // executed successfully.
-        if cfg!(windows) {
-            sock.bind(&inaddr_any(addr))?;
-        }
+        #[cfg(windows)]
```

Comment:
The change from `if cfg!(windows) { ... }` to `#[cfg(windows)]` is probably better, but seems unrelated to adding Redox support.